core/src/org/javarosa/core/model/condition/EvaluationContext.java

```diff
@@ -348,11 +349,19 @@ private void expandReferenceAccumulator(TreeReference sourceRef, DataInstance so
                         //Just by getting here we're establishing a position for evaluating the current
                         //context. If we break, we won't push up the next one
                         positionContext[predIndex]++;
+                        boolean passed = false;
 
                         //test the predicate on the treeElement
                         //EvaluationContext evalContext = new EvaluationContext(this, treeRef);
                         EvaluationContext evalContext = rescope(treeRef, positionContext[predIndex]);
-                        Object o = xpe.eval(sourceInstance, evalContext);
+                        Object o;
+                        try {
+                            o = xpe.eval(sourceInstance, evalContext);
+                        } catch (XPathTypeMismatchException e) {
+                            // Yay, the predicate's so jacked it failed to evaluate!
+                            passedAll = false;
```

Comment:
Hey,

So we don't "fail" here quite yet. We only "fail" on this output if we actually try to _Access_ the nodeset. If, for instance, all we're doing is Count()ing this nodeset, we intentionally don't fail, because that can be used as a way to shield references that would otherwise crash due to unavailable data.

for example:

if(count(instance('instance')/data/node/that/may/not/exist) = 0, '', instance('instance')/data/node/that/may/not/exist)

I think we might actually need this exception to bubble all the way up for that functionality to work, thoug?